eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/GoodLoggingCheck.java

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FullIdent;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.TokenUtil;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Good Logging Practice:
+ * <ol>
+ * <li>Non-static Logger instance.</li>
+ * <li>Logger variable name should be 'logger' in public class and not exist in an implementation package.</li>
+ * <li>Should not use any external logger class, only use ClientLogger. No slf4j, log4j, or other logging imports are allowed.</li>
+ * <li>'System.out' and 'System.err' is not allowed as well.</li>
+ * </ol>
+ */
+public class GoodLoggingCheck extends AbstractCheck {
+    private static final String CLIENT_LOGGER_PATH = "com.azure.core.util.logging.ClientLogger";
+    private static final String CLIENT_LOGGER = "ClientLogger";
+    private static final String LOGGER_NAME_ERROR = "ClientLogger instance naming: use ''%s'' instead of ''%s'' for consistency.";
+    private static final String STATIC_LOGGER_ERROR = "Reference to ClientLogger should not be static: remove static modifier.";
+    private static final String NOT_CLIENT_LOGGER_ERROR = "Do not use external logger class. Use ''%s'' as a logging mechanism instead of ''%s''.";
+
+    private boolean hasClientLoggerImported;
+    private String loggerRequiredName;
+    private String className;
+
+    private static final Set<String> INVALID_LOG_SET = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(
+        "org.slf4j", "org.apache.logging.log4j"
+    )));
+
+    /**
+     * Setter to specifies valid identifiers
+     * @param loggerRequiredName the variable name of logger used
+     */
+    public void setLoggerName(String loggerRequiredName) {
```

Comment:
Where is this set? Maybe some documentation if it is set in configuration files?

Reply:
This is another way to add property values to the checkstyle rule:     
```
<module name="com.azure.tools.checkstyle.checks.GoodLoggingCheck">
      <property name="loggerName" value="logger"/>
 </module>
```

setLoggerName method will set the value of loggerRequiredName as 'logger'

Reply:
This is custom property setup. But I will change back to use java code to do the same thing.